llama_stack/apis/eval/eval.py

```diff
@@ -61,15 +61,42 @@ class EvaluateResponse(BaseModel):
     scores: Dict[str, ScoringResult]
 
 
+@json_schema_type
+class EvalJob(CommonJobFields):
+    """
+    An evaluation job.
+
+    :param type: The type of the job.
+    :param result_files: The file ids of the eval results.
+    :param result_datasets: The ids of the datasets containing the eval results.
+    :param benchmark_id: The id of the benchmark to evaluate on.
+    :param candidate: The candidate to evaluate on.
+    """
+
+    type: JobType = JobType.evaluation.value
```

Comment:
Should this be `Literal[JobType.evaluation.value]` maybe? The way the model is defined at the moment, I think it will potentially allow to construct an `EvalJob` with a different type.

Reply:
> I think it will potentially allow to construct an EvalJob with a different type.

Hmm, could you give an example on a different EvalJob?

Reply:
I mean that the model class will allow to create a model object that would have the `type` attribute set to something that is different from `evaluation.value`. I'd prefer if it doesn't allow it on the type system level. (Of course, one may argue that since we are the ones that construct the job object, we may just not write buggy code. Alas, we do; and we don't fully control what providers do - especially once we implement out-of-tree providers.)